src/profile-logic/transforms.js

```diff
@@ -449,6 +488,43 @@ function _dropFunctionInCallNodePath(
   return callNodePath.includes(funcIndex) ? [] : callNodePath;
 }
 
+// removes all functions that are not in the category from the callNodePath
+function _removeOtherCategoryFunctionsInNodePathWithFunction(
+  thread: Thread,
+  category: IndexIntoCategoryList,
+  callNodePath: CallNodePath
+): CallNodePath {
+  const catsPerFunc = getCategoriesPerFunction(thread);
+  const inCategory = callNodePath.map((n) => catsPerFunc[n].includes(category));
+  return callNodePath.filter((n, i) => {
+    if (inCategory[i]) {
+      return true;
+    }
+    return false;
+  });
+}
```

Comment:
I'm sorry to insist here, but 1. this is incorrect (this will keep functions that shouldn't be kept) and 2. this isn't performant (there are a lot of loops -- that one could be improved if `catsPerFunc` used `Set` though, but we'd still have a lot of expensive loops in `_getCategoriesPerFunction`).

We really have to work with the call node table to make it correct. (Note: I thought about using the stack table, but unfortunately this wouldn't work easily, because we can have more than 1 stacks with the same prefix and function in the stack table...).

There's no easy way to get it, besides putting the call node table in the action. You can get it and inject it in the action in https://github.com/firefox-devtools/profiler/blob/a69c98867d185c069560c5b1aa56da8c61ac97b9/src/actions/profile-view.js#L1883-L1892, then pass it to `applyTransformToCallNodePath` when it's called in `reducers/profile-view.js`.

Then here is some pseudo code for this function:
```js
const newCallNodePath = [];

let prefix = -1;
for (const funcIndex of callNodePath) {
  const callNodeIndex = _getCallNodeIndexFromParentAndFunc(prefix, funcIndex, callNodeTable);
  if (callNodeIndex === null) {
    throw new Error(`We couldn't find a node with prefix ${prefix} and func ${funcIndex}, this shouldn't happen.`);
  }

  if (callNodeTable.category[callNodeIndex] === category) {
    newCallNodePath.push(funcIndex);
  }

  prefix = callNodeTable.prefix[callNodeIndex];
}

return newCallNodePath;
```

And the function can be renamed `_filterCategoryFromPath` for example.

As you see, that's just one loop over the call node table at maximum.

Reply:
It didn't came to me that I could just pass the callNodeTable through.

nit: `prefix = callNodeTable.prefix[callNodeIndex];` -> `prefix = callNodeIndex;`

Reply:
> nit: prefix = callNodeTable.prefix[callNodeIndex]; -> prefix = callNodeIndex;

oops that's right